js_interop/lib/src/date.dart

```diff
@@ -0,0 +1,314 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:js_interop';
+
+/// The JS [`Date`] type.
+///
+/// [`Date`]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date
+@JS('Date')
+extension type JSDate._(JSObject _) implements JSObject {
+  /// The [Date constructor] that returns the current date and time.
+  ///
+  /// [Date constructor]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date
+  external JSDate.now();
+
+  /// The [Date constructor] with the number of milliseconds since the epoch of
+  /// January 1, 1970, UTC.
+  ///
+  /// [Date constructor]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date#time_value_or_timestamp_number
+  external JSDate.fromMillisecondsSinceEpoch(int millisecondsSinceEpoch);
+
+  /// The [Date constructor] that parses its value [from a string].
+  ///
+  /// [Date constructor]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date
+  /// [from a string]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date#date_string
+  external JSDate.parse(String dateString);
+
+  /// The [Date constructor] that copies its value [from an existing Date].
+  ///
+  /// [Date constructor]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date
+  /// [from an existing Date]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date#date_object
+  external JSDate.from(JSDate other);
+
+  /// The [Date constructor] that uses [individual component integers].
+  ///
+  /// [Date constructor]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date
+  /// [individual component integers]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date#individual_date_and_time_component_values
+  external JSDate(int year, int month,
+      [int? day, int? hours, int? minutes, int? seconds, int? milliseconds]);
+
+  /// Dee [`Date.now()`].
+  ///
+  /// [`Date.now()`]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/now
+  @JS('now')
+  external static int nowAsMillisecondsSinceEpoch();
+
+  /// See [`Date.parse()`].
+  ///
+  /// [`Date.parse()`]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse
+  @JS('parse')
+  external static int parseAsMillisecondsSinceEpoch(String dateString);
+
+  /// The [`Date.utc()`] function.
+  ///
+  /// [`Date.utc()`]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/UTC
+  @JS('UTC')
+  external static int utc(int year,
```

Comment:
We avoid renaming to lowercase with `package:web` e.g. `URL`, maybe worth doing the same?

Same comment on `toIsoString`, `toUtcString`, and `toDartUtc`.